cs130_front/src/components/ListingCreator/ListingCreator.js

```diff
@@ -0,0 +1,67 @@
+import React, { useState } from 'react';
+import axios from 'axios';
+import { withRouter } from 'react-router-dom';
+import TextArea from '../TextInput/TextArea';
+import Text from '../Text/Text';
+import Button from '../Button/Button';
+import { USER_SERVER } from '../../Config';
+import './ListingCreator.css'
+import Dropdown from '../Dropdown/Dropdown';
+
+function ListingCreator(props) {
+  const [message, setMessage] = useState('');
+  const[group, setGroup] = useState(props.items[0]);
+  const noGroupOption = {name: "None",groupId: 0};
+
+  async function createListing() {
+    try {
+      const dataToSubmit = {
+        poster: props.user,
+        course_name: props.course_name,
+        course_id: 0, // TODO, get courseID from course
+        text_description: message,
+        group_id: 0, // TODO, get groupID from group
+      }
+      console.log(dataToSubmit)
+      const response = await axios.post(`${USER_SERVER}/classes/addListing`,  dataToSubmit)
```

Comment:
for the future, this endpoint is under course.
```
	courseRoutes := routes.PathPrefix("/course").Subrouter()
	courseRoutes.HandleFunc("/addListing", a.handleRequest(handlers.CreateListing)).Methods("POST")
```
so you would need to do add a COURSE_SERVER on Config,js similarly to how USER_SERVER is there now but replace user with course